### DISCUSSION TITLE
Switch Between Multiple Roles

### DISCUSSION THREAD

**Discussion by philkobby** [spatie/laravel-permission#2061] [Q&A]
Assuming that I have 3 roles assigned to the same user.

admin
dashboard = admin

sales
dashboard = sales

marketing
dashboard = marketing

how can I switch between these three roles to get redirected to the preferred dashboard?

**Comment by erikn69** [comment]
Try with a `switch` or an `if`
```php
$user = auth()->user();
switch(true){
    case $user->hasRole('admin'):
        // redirect admin
    case $user->hasRole('sales'):
        // redirect sales
    case $user->hasRole('marketing'):
        // redirect marketing
}
```

**Reply by philkobby** [reply]
Thank you,
did that but always return the first case since the user has the admin role too
is there a way for the authenticated user to specify the role to be used in the case?

i.e Switch role to either
admin
sales 
marketing

**Reply by erikn69** [reply]
Just sort by relevance, you can change as you need, 
>is there a way for the authenticated user to specify the role to be used in the case?

use links with routes for user select wich dashboard want to use, also you can save on db some config for always start with the same selected dashboard as default

Mark this as answered if it works for you

**Reply by philkobby** [reply]
noted
Thank you
